### PR TITLE
feat(tide): support tide nodepool

### PR DESCRIPTION
**Pull request by xjh1996**
#### What type of PR is this?

https://github.com/kubewharf/katalyst-core/issues/213
#### What this PR does / why we need it:
Support tide node pool in katalyst. When a workload is scaled down, the vacant nodes will be identified and assigned to another workload that is experiencing higher demand. This will facilitate the efficient utilization of nodes and prevent resource wastage.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -34,21 +34,23 @@ require (
 	golang.org/x/sys v0.7.0
 	golang.org/x/time v0.0.0-20220210224613-90d013bbcef8
 	google.golang.org/grpc v1.51.0
-	k8s.io/api v0.24.6
-	k8s.io/apimachinery v0.24.6
-	k8s.io/apiserver v0.24.6
-	k8s.io/client-go v0.24.6
-	k8s.io/component-base v0.24.6
-	k8s.io/component-helpers v0.24.6
+	k8s.io/api v0.24.16
```
Comment (waynepeking348) [inline go.mod:37]: is it inevitable to upgrade go mod dependency for k8s-related repo?

Comment (waynepeking348) [reply go.mod:37]: mainly considered for stability, prefer not to do it if possible

File: pkg/controller/tide/tide.go [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+		if !selector.Matches(labels.Set(node.Node().GetLabels())) {
+			continue
+		}
+		nodeUsageList = append(nodeUsageList, NodeUsage{
+			node:    node.Node(),
+			usage:   nodeutil.NodeUtilization(pods, resourceNames),
+			allPods: pods,
+		})
+	}
+	// 节点按照 使用率排序
```
Comment (waynepeking348) [inline pkg/controller/tide/tide.go:701]: please change to english version for comments

Comment (waynepeking348) [reply pkg/controller/tide/tide.go:701]: the same for other comments

File: go.mod (hunk 1 above)
Comment (xjh1996) [reply go.mod:37]: This is triggered by the necessary dependency cluster-autoscaler. At the bottom of the go mod, there is a replace directive. Theoretically, the version used is still v0.24.6.

File: pkg/controller/tide/tide.go (hunk 2 above)
Comment (xjh1996) [reply pkg/controller/tide/tide.go]: done

**Review by waynepeking348 — APPROVED** [review]

**Comment by waynepeking348** [comment]
lgtm @caohe pls have a look for this pr

File: pkg/controller/tide/tide.go [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+				ti = ti + nodeUsageList[i].usage[name].Value()
+				tj = tj + nodeUsageList[j].usage[name].Value()
+			}
+		}
+		return ti < tj
+	})
+	return nodeUsageList, nil
+}
+
+// clearUnexpectedCNR is used to clear unexpected cnr
```
Comment (caohe) [inline pkg/controller/tide/tide.go:716]: It seems that this comment may be not relevant to this method.

**Comment by caohe** [comment]
Could you add more information to the description part?

File: pkg/controller/tide/tide.go (hunk 3 above)
Comment (xjh1996) [reply pkg/controller/tide/tide.go]: done

**Comment by xjh1996** [comment]
> Could you add more information to the description part?
@caohe done

**Review by waynepeking348 — APPROVED** [review]

**Review by caohe — APPROVED** [review]

**Pull request merged**